src/charm.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+"""GLAuth Operator Charm."""
+
+import logging
+
+import glauth
+from charms.operator_libs_linux.v1 import snap
+from ops.charm import CharmBase
+from ops.main import main
+from ops.model import ActiveStatus, BlockedStatus, MaintenanceStatus
+
+logger = logging.getLogger(__name__)
+
+
+class GlauthCharm(CharmBase):
+    """Charmed Operator to deploy glauth - a lightweight LDAP server."""
+
+    def __init__(self, *args):
+        super().__init__(*args)
+        self.glauth = glauth
+
+        # Observe common Juju events
+        self.framework.observe(self.on.install, self._install)
+        self.framework.observe(self.on.remove, self._remove)
+        self.framework.observe(self.on.update_status, self._update_status)
+        self.framework.observe(self.on.upgrade_charm, self._upgrade_charm)
+
+        # Integrations
+        self.framework.observe(self.on.glauth_relation_changed, self._on_glauth_relation_changed)
+
+    def _install(self, _):
+        """Install glauth."""
+        self.unit.status = MaintenanceStatus("installing glauth")
+        try:
+            self.glauth.install()
+            self.unit.set_workload_version(self.glauth.version)
+            self.unit.status = ActiveStatus("glauth ready")
+        except snap.SnapError as e:
+            self.unit.status = BlockedStatus(str(e))
```

Comment:
Hmm... I do not think setting the unit status to `BlockedStatus(str(e))` is exactly what we want here. Calling the `str` builtin on `snap.SnapError` will return the string version of the exception. In SnapError's case, it will be converted to the string returned by SnapError's `__repr__` method:

```python3
def __repr__(self):
    """Represent the Error class."""
    return "<{}.{} {}>".format(type(self).__module__, type(self).__name__, self.args)
```

This means that the human operator (person looking at `juju status`) will see `<charms.operator_libs_linux.v1.snap.SnapError "all args passed to SnapError when raised">` rather than just the error message. I am wondering if this might scare someone who is not familiar with Python programming. Instead, let's just the message of the SnapError as the status:

```suggestion
            self.unit.status = BlockedStatus(e.message)
```

Now if a SnapError is raised like `SnapError("Uh oh spaghetti Os")`, the blocked status message will be set to `"Uh oh spaghetti Os"` rather than the string representation of the exception.